### DISCUSSION TITLE
Newbie question, why I am getting an exception when creating an appender (C++)

### DISCUSSION THREAD

**Discussion by RallyTronics** [duckdb/duckdb#6694] [Q&A]
I am using VS2022, 64 bit, Windows 11.  ( using libduckdb-windows-amd64.zip )

My code:
```
    duckdb::DuckDB db("test.db");
    duckdb::Connection con(db);
    // create the table
    con.Query("CREATE TABLE people(id INTEGER, name VARCHAR)");
    // initialize the appender
    duckdb::Appender appender(con, "people");
```
I get an unhandled exception when trying to instantiate the Appender

![image](https://user-images.githubusercontent.com/7726730/224579653-5ed0c21f-77dd-4883-9225-3555092505fb.png)

Any thoughts?

**Comment by RallyTronics** [comment]
Some additional information, the same test using the C API works fine.

**Comment by Mytherin** [comment]
That code snippet works fine for me. You are ignoring the result of the table-creation, however. Perhaps check if something went wrong there?

**Comment by RallyTronics** [comment]
My updated code:
```
    duckdb::DuckDB db("test.db");
    duckdb::Connection con(db);
    // create the table
    std::unique_ptr<duckdb::MaterializedQueryResult> result = con.Query("CREATE TABLE people(id INTEGER, name VARCHAR)");
    if (result->HasError())
    {
        const std::string err(result->GetError());
        std::cout << err << "\n";
    }

    // initialize the appender
    duckdb::Appender appender(con, "people");

```
result->HasError is true but the error reported is not helpful.  It returns "rror:"  Am I checking the return value correctly?

![image](https://user-images.githubusercontent.com/7726730/224716306-01c7e833-113a-4a9e-8f3c-029d2b9e26ac.png)
